### PR TITLE
Add option to show diagnostics in a popup

### PR DESCRIPTION
**Pull request by D4yvid**
Added a option to show diagnostics in a floating popup

image

The option name is `showDiagInPopup` and it's default is `false`

### CODE REVIEW

File: autoload/lsp/options.vim [hunk 1]
```diff
@@ -33,7 +33,9 @@ export var lspOptions: dict<any> = {
   # Show a diagnostic message on a status line
   showDiagOnStatusLine: false,
   # Don't print message when a configured language server is missing.
-  ignoreMissingServer: false
+  ignoreMissingServer: false,
+  # Make diagnostics show in a popup instead of echoing
+  showDiagInPopup: true
```
Comment (yegappan) [inline autoload/lsp/options.vim:38]: Can you add also update the lsp.txt help file with a description of this new option?

Comment (D4yvid) [reply autoload/lsp/options.vim:38]: Yeah, i'll do that

Comment (D4yvid) [reply autoload/lsp/options.vim:38]: Done, i've added

File: doc/lsp.txt [hunk 2]
```diff
@@ -264,6 +264,9 @@ showDiagOnStatusLine	Show a diagnostic message on a status line.
 			By default this is set to false.
 ignoreMissingServer	Do not print a missing language server executable.
 			By default this is set to false.
+showDiagInPopup		Use a popup to display server diagnostics instead of
+			echoing.
+			By default this is set to false.
```
Comment (yegappan) [inline doc/lsp.txt:269]: By default, is this option set to true or false?

Comment (D4yvid) [reply doc/lsp.txt:269]: Wait, i've forgot to change it back to false, i was testing it

Comment (D4yvid) [reply doc/lsp.txt:269]: Sorry, i've changed it back to false

**Pull request merged**